src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -527,6 +527,13 @@ public static bool IsNullOrWhiteSpace([NotNullWhen(false)] string? value)
         internal ref char GetRawStringData() => ref _firstChar;
         internal ref ushort GetRawStringDataAsUInt16() => ref Unsafe.As<char, ushort>(ref _firstChar);
 
+        internal static string AllocateInternal(int length, out Span<char> resultSpan)
+        {
+            string result = FastAllocateString(length);
+            resultSpan = new Span<char>(ref result._firstChar, result.Length);
+            return result;
+        }
```

Comment:
Should we rather use `string.Create` everywhere?

Reply:
If the delegate can be inlined everywhere, otherwise there may be performance hot spots?